datastore/query.go

```diff
@@ -45,22 +45,33 @@ const (
 	keyFieldName = "__key__"
 )
 
-var operatorToProto = map[operator]pb.PropertyFilter_Operator{
-	lessThan:    pb.PropertyFilter_LESS_THAN,
-	lessEq:      pb.PropertyFilter_LESS_THAN_OR_EQUAL,
-	equal:       pb.PropertyFilter_EQUAL,
-	greaterEq:   pb.PropertyFilter_GREATER_THAN_OR_EQUAL,
-	greaterThan: pb.PropertyFilter_GREATER_THAN,
-	in:          pb.PropertyFilter_IN,
-	notIn:       pb.PropertyFilter_NOT_IN,
-	notEqual:    pb.PropertyFilter_NOT_EQUAL,
-}
-
-// filter is a conditional filter on query results.
-type filter struct {
-	FieldName string
-	Op        operator
-	Value     interface{}
+var operatorToProto = map[string]pb.PropertyFilter_Operator{
+	lessThan.String():    pb.PropertyFilter_LESS_THAN,
+	lessEq.String():      pb.PropertyFilter_LESS_THAN_OR_EQUAL,
+	equal.String():       pb.PropertyFilter_EQUAL,
+	greaterEq.String():   pb.PropertyFilter_GREATER_THAN_OR_EQUAL,
+	greaterThan.String(): pb.PropertyFilter_GREATER_THAN,
+	in.String():          pb.PropertyFilter_IN,
+	notIn.String():       pb.PropertyFilter_NOT_IN,
+	notEqual.String():    pb.PropertyFilter_NOT_EQUAL,
+}
+
+func (op operator) String() string {
+	var operatorToString = map[operator]string{
```

Comment:
This map will be allocated on every call to string. Shall we move it up to package-level scope to avoid this?

Reply:
Moved